### DISCUSSION TITLE
Leantime — Roadmapping future development

### DISCUSSION THREAD

**Discussion by gloriafolaron** [Leantime/leantime#744] [Q&A]
We had a great comment on a thread here suggesting creating a product roadmap so that folks could go in and pick things more easily to contribute / work on.  Along with that usefulness, this struck us as a great idea because it creates a layer of clear direction, visibility and things to look forward to.

Probably most importantly, it gives credit and support to things y’all are working on and needing.

that said, while I intend to sort through the feature requests as I can — I was hoping to start some discussion around prioritizing features.

Feel free to copy and paste and fill in the below: 

1. If you had to pick 3 things that don’t currently exist but would help take Leantime to the next level for your workflow, what would they be with 1 being the most important and 3 being the least? 

—- 1. 
—- 2.
—- 3. 

2. Could you name: 1 feature that exists that you couldn’t live without, 1 feature that you wish would go away, and 1 feature you wish we’d get to updating already?

—— Can’t live without:
—— Kick to the curb:
—— Make it better:

3. What are you currently using leantime to manage?

4. Now the most important one… Star Wars or Star Trek?

**Comment by phulstaert** [comment]
Hi @gloriafolaron , 

Next to features, it would make sense to also plan bugs/issues in the roadmap.
High prio issues (security related), will be done in a small minor release, but it would make sense that there also could be low prio bugs that we chose not to solve in the next release, but plan on a later date. (because that part is getting a total overhaul or something like that).

For me personally, 
1. Make this project PSR-12 compliant.
2. Fix the err 500 on no LEAN_EMAIL_RETURN.
   a - Quick fix) first fix the documentation (i already did the docker command, but forgot the "Installation (Production)" section.
   b - real fix) something like "if the required fields for mail are not entered => disable mailing (+ write in logfile what is missing)"
3. make an online demo (using github.io?)

And finally... definitely Star Trek. :-)

**Reply by phulstaert** [reply]
Also, i see that the docker image of Leantime uses php 7.2; that is EOL since more than a year. Maybe time to test if php 8.1 works well. (https://www.php.net/supported-versions.php)

Maybe it would make sense to keep a list of used versions of tools/services/libraries and their end of support dates.

**Comment by marcelfolaron** [comment]
I have already a fix for the email issue in my bugFix branch. Will get that
in soon.

I’ll need some help with the PSR-12 compliance… there is a lot :P

I always thought GitHub.io was for static sites. I‘ll Check it out again.

**Reply by phulstaert** [reply]
I think you are right about the static part... I was looking at the demo site of https://planka.app and it 'felt' like a php site.
Maybe we could do an html dump of Leantime so people can at least get a taste of it.

**Comment by funkywaddle** [comment]
One feature I wish LeanTime had was accepting webhooks from git repo applications. 

I know it can send webhooks out to Discord, Slack, etc. But, it would be awesome if we could implement accepting webhooks. 

Receiving webooks from git repo applications would allow branch creations to move tickets to "working on", PR creations to move tickets to "In review". PR merges to move tickets to "Done", etc.

 I am willing to work on this myself, but am currently having issues installing the local/dev instance (see [here](https://github.com/Leantime/leantime/discussions/792)). Once my installation issue is resolved, I can start implementing this.

**Reply by marcelfolaron** [reply]
That would be awesome. Before you start would you mind sharing some thoughts on architecture? 
feel free to join our discord server to chat

**Comment by jmourinho** [comment]
Hi, 

i am totally new to Leantime. So please forgive me if something that i say is not accurate. I've been testing it and thinking of implementing it at my department in the company (innovation department). 

If you had to pick 3 things that don’t currently exist but would help take Leantime to the next level for your workflow, what would they be with 1 being the most important and 3 being the least?
—- 1. A full fledged Gantt chart (WP's, activities/subactivities, deliverables and milestones) instead of only milestones.
—- 2. Integration with Microsoft Teams
—- 3. 

Could you name: 1 feature that exists that you couldn’t live without, 1 feature that you wish would go away, and 1 feature you wish we’d get to updating already?
—— Can’t live without: (Too soon to tell)
—— Kick to the curb:  (Too soon to tell)
—— Make it better: (Too soon to tell)

What are you currently using leantime to manage? Innovation Projects

Now the most important one… Star Wars or Star Trek? Star Wars :)

**Reply by funkywaddle** [reply]
I'm not much of a MS Teams user, but do you know if they accept webhooks, similar to how Discord does? It might be pretty easy to add that, if they do.

**Reply by phulstaert** [reply]
Could you elaborate on "Integration with Microsoft Teams"? Do you mean integration with Microsoft Planner, Microsoft ToDo, Microsoft Project,... or do you mean importing the users/teams or maybe easy linking to SharePoint files (Each "team "is in reality a SharePoint site, so shared files (even in the chat) are actually on a SharePoint site.)
Or did you mean something else? And from all those features, what is the most important to you?

I'm not sure, but a JSON link you could use (instead of the export), so you could link it in PowerBI and generate your Gantt charts (and and much more), would that be interesting? (just thinking out loud here)

edit: typos

**Reply by jmourinho** [reply]
By "integration with MS teams" i mean:

- Integration with planner (where what you do in one side is reflected in the othe Leantime Kanban<->MS Planner)
- Association of conversations/chat from teams
- Association of a LeantimeProject with a MS Teams "team"

But without any doubt, the top feature that would convince me to adopt leantime in our department would be the full fledged gantt. Tha would be great.

**Reply by ckvtz** [reply]
Hello,

I am totally new to Leantime as well. Great APP and thank you for the community version. I too think full fledged gantt would be the top feature!

**Comment by marcelfolaron** [comment]
Microsoft has a Slack alternative called Microsoft Teams (
https://www.microsoft.com/en-us/microsoft-teams/group-chat-software) I
assume it's related to that.
They do have webhooks but I haven't looked further into the structure.
Should be fairly simple though. Now, who has an MS Team account :D

**Reply by funkywaddle** [reply]
unfortunately, I do. But I'm not willing to admit it. I had to have one for a job interview. Haven't touched it since. lmao

**Reply by jmourinho** [reply]
Well, i've used slack in the past, but MS Teams is just so integrated into the MS ecosystem that it is hard to escape from it. Slack is a bit better, but MS Teams compensates that with great integration, conference features, deep integration with MS office, etc.

Disclaimer: i'm not anything's fanboy. I just use what rationally makes sense for me.

**Comment by joaobrazuna** [comment]
—- 1. Integration with GIT  (Can’t live without)
—- 2. Integration with MSTeams (Make it better)
—- 3. Simple Integration with AD/LDAP like https://gitea.io/ has. (Make it better)

What are you currently using leantime to manage?
- Sofware projects

Now the most important one… Star Wars or Star Trek?
- Stargate SG1 :-)

**Reply by marcelfolaron** [reply]
Question on the git integration. What exactly would you like to see in there?

**Reply by joaobrazuna** [reply]
Some think like this:
https://www.openproject.org/docs/system-admin-guide/integrations/github-integration/
I do not consider this as a "must have" but more like a "it would be nice"

Another thing that would be VERY nice  is the possibility to export the "Gantt view" 
Now I have to copy the screen if I want to include the Gantt in a report.

![image](https://user-images.githubusercontent.com/44297535/182591296-7aacd069-686e-4175-9aef-bbfd8ebefa3d.png)

**Comment by marcelfolaron** [comment]
Tek'ma'te my friend!

I will take a look at giteas ldap integration and see if I can replicate
that.

**Comment by ckvtz** [comment]
Hello,

I am totally new to Leantime as well. Great APP and thank you for the community version. I too think full fledged gantt would be a top feature! As mentioned by jmourinho: A full fledged Gantt chart (WP's, activities/subactivities, deliverables and milestones) instead of only milestones.

**Reply by marcelfolaron** [reply]
Thanks for the feedback. 
Couple questions:
Where would you prefer to have that Gantt chart? As part of the milestone view (maybe a checkbox to view tasks + milestones)?

Or would you like to see that as part of the To-Do section. A separate view like Table/Kanban. I am already considering putting in a calendar view for the tasks, so gantt would fit as well.

The following up question on that is: Is the individual milestone view (as its own menu point) even valuable? Or would you prefer to see everything within the To-Do section?

**Reply by jmourinho** [reply]
Thank you for the responsiveness. 

I would suggest that the "milestones" word in the left menu bar could be called "Gantt", while generally keeping the same "Milestones" layout. The button "Add Milestone" could be replaced by a dropdown button:

"Add"
-> Work Package
-> Task
-> Deliverable
-> Milestone

As a Gantt can include WPs, Tasks, milestones and deliverables that would be also a painless solution also for current users (if they do not need the WPs and tasks and deliverables, they just don't add them to the chart). They could create a gantt only with milestones as they do now, for example.

**Reply by martin-braun** [reply]
@marcelfolaron I haven't installed Leantime yet, but reading about the features shows me that a kanban view is only available for projects, but not to-do's, right? Is this correct?

I think Leantime should cover the basic features of Trello, this would make it a killer product. :)

**Comment by ckvtz** [comment]
You are welcome.

I agree, this would be easy for people who are already used to the UX of
this application and the dev team as well.

I have another suggestion as well. I suppose you currently do not allow the
single right label field attribute in the gantt to be configurable. It
would be great feature if you could allow a configurable attribute field
for example: Start Date, Finish Date, Assignee, Subject, Progress% etc.,

You could also think of having a LEFT, RIGHT and FAR RIGHT LABEL which
would give more context to the Gantt chart view.

**Comment by marcelfolaron** [comment]
Thank you for the additional context! I think that makes sense
@ckvtz can you explain a little more (or show a screenshot) of the label
you are referring to?
Are you referring to the text next to the title "(x% done)" and being able
to change what is displayed there?

**Comment by ckvtz** [comment]
Hello,

Please find Attached screenshots that I suppose are self explanatory. Pls
let me know if you have any questions.

Best Rgds,
Chandar.

**Comment by ckvtz** [comment]
@Marcel, did you find this helpful?

**Comment by marcelfolaron** [comment]
It doesn’t seem like the screenshots made it. But I believe I understand
what you are referring to.

**Comment by ckvtz** [comment]
By “It doesn’t seem like the screenshots made it” do you mean to say you
did not receive the attachments or that I did not answer your question
appropriately.

Always happy to provide any feedback if necessary. Glad to help.

Thank you.

**Comment by marcelfolaron** [comment]
I didn’t receive the attachment.

**Comment by ckvtz** [comment]
Attaching the same again. Please let me know if you get it or else a better
way for me to share the same.

Thank you.

**Comment by adocampo** [comment]
Just testing Leantime, so no deep knowledge at all

    If you had to pick 3 things that don’t currently exist but would help take Leantime to the next level for your workflow, what would they be with 1 being the most important and 3 being the least?

—- 1. Kanban complex subtask as dependence
—- 2. Incoming webhooks
—- 3. LDAP configuration via UI, and those features: sync (users and groups), filtering, and auto-mapping.

**Comment by joaobrazuna** [comment]
4-Integration with GIT


Melhores Cumprimentos,

João Brazuna
(Responsável Informática Interna)

Telefone: 213122921<tel:+351213122921>   |    Telemóvel: 915752312<tel:+351915752312>
Morada:   Rua Sarmento Beires 3A   1900-410 Lisboa
Website: https://www.beltraocoelho.pt<https://www.beltraocoelho.pt/>


________________________________
De: Angel Docampo ***@***.***>
Enviado: terça-feira, 25 de abril de 2023, 08:29
Para: Leantime/leantime ***@***.***>
Cc: João Brazuna ***@***.***>; Comment ***@***.***>
Assunto: Re: [Leantime/leantime] Leantime — Roadmapping future development (Discussion #744)


Just testing Leantime, so no deep knowledge at all

If you had to pick 3 things that don’t currently exist but would help take Leantime to the next level for your workflow, what would they be with 1 being the most important and 3 being the least?


—- 1. Kanban complex subtask as dependence
—- 2. Incoming webhooks
—- 3. LDAP configuration via UI, and those features: sync (users and groups), filtering, and auto-mapping.
